### DISCUSSION TITLE
How to interact with ComfyUI via API?

### DISCUSSION THREAD

**Discussion by mypeaceduck** [comfyanonymous/ComfyUI#4772] [Q&A]
Hi,

I have my web interface.

Field to upload image, field to enter prompt, field to show result.

<img width="1218" alt="Screenshot 2024-09-04 at 01 24 43" src="https://github.com/user-attachments/assets/b58f3893-aa30-45e2-82a0-6388a98b4ae2">

- In this example all fields are filled in ComfyUI interface: https://comfyanonymous.github.io/ComfyUI_examples/inpaint/

![image](https://github.com/user-attachments/assets/25012557-26e8-4d47-92cf-723925c74b92)

Is it possible to fill them from my web interface via API? 

![inpaint_example](https://github.com/user-attachments/assets/296b5cda-9316-4b30-b42a-abf112273a89)

**Comment by ltdrdata** [comment]
Is what you want a UI that is displayed in a completely separate tab?
Or is it a UI displayed on the ComfyUI workflow viewer?

One point to be careful about is that the workflow does not contain complete information to execute the prompt. When execution occurs, the workflow is processed into what's called a prompt through the serialize logic of each node.

And what's actually used for execution is this prompt.

**Reply by mypeaceduck** [reply]
Thank you very much for your reply.

> Is what you want a UI that is displayed in a completely separate tab? Or is it a UI displayed on the ComfyUI workflow viewer?

I want to make a web application for my users to work exactly like ComfyUI, but only through my interface (NextJS). Since ComfyUI interface is very overloaded and users are not used to it.

**Reply by ltdrdata** [reply]
> I want to make a web application for my users to work exactly like ComfyUI, but only through my interface (NextJS). Since ComfyUI interface is very overloaded and users are not used to it.

If your interface has a fixed form, you need to use a method where you extract the prompt in the form of an API export from the corresponding workflow, modify that prompt, and then send the API request.

**Reply by mypeaceduck** [reply]
> If your interface has a fixed form, you need to use a method where you extract the prompt in the form of an API export from the corresponding workflow, modify that prompt, and then send the API request.

I want my users to send an API request, but I don't see documentation ComfyUI on how to do this or if it's possible.

**Reply by ltdrdata** [reply]
When using ComfyUI as a backend, the official document you can refer to is the example code.
https://github.com/comfyanonymous/ComfyUI/tree/master/script_examples

**Reply by bigcat88** [reply]
> > If your interface has a fixed form, you need to use a method where you extract the prompt in the form of an API export from the corresponding workflow, modify that prompt, and then send the API request.
> 
> I want my users to send an API request, but I don't see documentation ComfyUI on how to do this or if it's possible.

Are we doing the same thing by any chance? :)

We did maybe something similar in this [repo](https://github.com/Visionatrix/Visionatrix) to what you are developing, we also had the initial goal of making ComfyUI flows with simple interfaces.

**Comment by Kahdeg-15520487** [comment]
so i captured this post request when i press queue prompt, i just need to send this to the comfyui url right?

![image](https://github.com/user-attachments/assets/760cf604-44ae-421f-b496-c9522cf1ce54)
